src/services/PollingService.ts

```diff
@@ -0,0 +1,50 @@
+import type { Disposable } from '../types';
+
+export type Runner = () => Promise<void>;
+
+/**
+ * Service that polls a function at a minimum interval.
+ */
+export class PollingService implements Disposable {
+  private _timeout?: NodeJS.Timeout;
+
+  private _isRunning = false;
+  get isRunning(): boolean {
+    return this._isRunning;
+  }
+
+  start = (run: () => Promise<void>, minInterval: number): void => {
+    this.stop();
+
+    this._isRunning = true;
+
+    const poll = async (): Promise<void> => {
+      const start = performance.now();
+
+      await run();
+
+      // Ensure checks don't run more often than the interval
```

Comment:
Need to check `isRunning` here as well to make sure we need to set the next poll.

Reply:
Done